dwave/optimization/include/dwave-optimization/array.hpp

```diff
@@ -723,94 +746,107 @@ class Array {
-    // fields. This can be any location within the underlying physical memory
-    // block of the exporter. For example, with negative strides the value may
-    // point to the end of the memory block.
-    // For contiguous arrays, the value points to the beginning of the memory
-    // block.
+    /// A pointer to the start of the logical structure described by the buffer
+    /// fields. This can be any location within the underlying physical memory
+    /// block of the exporter. For example, with negative strides the value may
+    /// point to the end of the memory block.
+    /// For contiguous arrays, the value points to the beginning of the memory
+    /// block.
     virtual double const* buff(const State& state) const = 0;
 
-    // product(shape) * itemsize. For contiguous arrays, this is the length of
-    // the underlying memory block. For non-contiguous arrays, it is the length
-    // that the logical structure would have if it were copied to a contiguous
-    // representation.
+    /// For contiguous arrays, this is the length of the underlying memory block.
+    /// For non-contiguous arrays, it is the length that the logical structure
+    /// would have if it were copied to a contiguous representation.
+    /// If the array is dynamic, returns Array::DYNAMIC_SIZE.
     ssize_t len() const { return (size() >= 0) ? size() * itemsize() : DYNAMIC_SIZE; }
+
+    /// For contiguous arrays, this is the length of the underlying memory block.
+    /// For non-contiguous arrays, it is the length that the logical structure
+    /// would have if it were copied to a contiguous representation.
+    /// Always returns a positive number.
     ssize_t len(const State& state) const { return size(state) * itemsize(); }
 
-    // Exactly sizeof(double)
+    /// Exactly ``sizeof(double)``.
     constexpr ssize_t itemsize() const { return sizeof(double); }
 
-    // "d" for double, see https://docs.python.org/3/library/struct.html
+    /// "d" for double, see https://docs.python.org/3/library/struct.html
     const std::string& format() const;
 
-    // The number of dimensions the memory represents as an n-dimensional array.
-    // If 0 the buffer points to a single item representing a scalar
+    /// The number of dimensions the memory represents as an n-dimensional array.
+    /// If 0 the buffer points to a single item representing a scalar.
     virtual ssize_t ndim() const = 0;
 
-    // An array of ndim length indicating the shape of the buffer contents
-    // as an n-dimensional array.
-    // Note that this is in terms of the actual type rather than in terms of
-    // num bytes.
-    // If the shape is state-dependent, the first value in shape will be
-    // DYNAMIC_SIZE
+    /// An array of ndim length indicating the shape of the buffer contents
+    /// as an n-dimensional array.
+    /// Note that this is in terms of the actual type rather than in terms of
```

Comment:
This comment is unclear and "num" should be either "number of" or ``num``

Applies below too